src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/tools/GoldPan.java

```diff
@@ -160,16 +179,49 @@ public void updateRandomizer() {
 
     @Override
     public @Nonnull List<ItemStack> getDisplayRecipes() {
-        List<ItemStack> recipes = new LinkedList<>();
+        List<ItemStack> recipes = new ArrayList<>();
 
         for (GoldPanDrop drop : drops) {
-            if (drop.getValue() > 0) {
-                recipes.add(new ItemStack(getInputMaterial()));
+            if (drop.getValue() <= 0) {
+                continue;
+            }
+
+            for (Material material : getInputMaterials()) {
+                recipes.add(new ItemStack(material));
                 recipes.add(drop.getOutput());
             }
         }
 
         return recipes;
     }
 
+    /**
+     * This returns whether the {@link GoldPan} accepts the {@link ItemStack} as an input
+     *
+     * @param itemStack
+     *            The {@link ItemStack} to check
+     *
+     * @return If the {@link ItemStack} is valid
+     */
+    public boolean isValidInput(ItemStack itemStack) {
+        if (itemStack == null) {
+            return false;
+        }
+
+        Material material = itemStack.getType();
+        return isValidInputMaterial(material) && SlimefunUtils.isItemSimilar(itemStack, new ItemStack(material), true, false);
+    }
+
+    /**
+     * This returns whether the {@link GoldPan} accepts the {@link Material} as an input
+     *
+     * @param material
+     *            The {@link Material} to check
+     *
+     * @return If the {@link Material} is valid
+     */
+    public boolean isValidInputMaterial(Material material) {
```

Comment:
```suggestion
    public boolean isValidInputMaterial(@Nonnull Material material) {
```

Goign to take a stab this is nonnull, though check and if from context it can be nullable, annotate it as such instead.